x-pack/plugins/security/public/plugin.ts

```diff
@@ -28,18 +43,48 @@ export class SecurityPlugin implements Plugin<SecurityPluginSetup, SecurityPlugi
     this.sessionTimeout = new SessionTimeout(notifications, sessionExpired, http, tenant);
     http.intercept(new SessionTimeoutHttpInterceptor(this.sessionTimeout, anonymousPaths));
 
+    this.license$ = licensing.license$;
+
     return {
       anonymousPaths,
       sessionTimeout: this.sessionTimeout,
     };
   }
 
-  public start() {
+  public start(core: CoreStart) {
     this.sessionTimeout.start();
+
+    const securityLicenseService = new SecurityLicenseService().setup();
+    this.licenseSubscription = this.license$.subscribe(rawLicense => {
```

Comment:
question: should we add the ability to subscribe to the `SecurityLicenseService`? @azasypkin what do you think about this?

Reply:
The initial idea was to use and set up this service only at the `plugin.ts` level and pass `license: SecurityLicense` property of its `setup` contract around. In this case `SecurityLicenseService` in `plugin.ts` handles license changes and `license` property is always up-to-date. That would allow us to only work with `SecurityLicense` and never leak Licensing Plugin setup / raw `ILicense` to anywhere else. 

If it's not enough now, then yeah, either `SecurityLicenseService.setup` should additionally expose `license$: Observable<SecurityLicense>` or existing `license` should have a property similar to `changes$`.

Or it's not what you're asking about?

Reply:
Yup! That's what I was asking. Thanks.